common/src/events/marshal/marshal.ts

```diff
@@ -39,7 +39,13 @@ export abstract class SprocketEventMarshal {
                         return;
                     }
 
-                    eventFunction(payload.data).catch(e => { this.logger.error(e) });
+                    try {
+                        // No need to catch; it's in a try-catch block because we don't know if it's a promise or not.
+                        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                        eventFunction(payload.data);
```

Comment:
If its async, you have to `await` the Promise in order for the error to be caught in the try/catch.

You should be able to use `instanceof` to determine if its a Promise or not

```ts
try {
    const maybePromise = someFunction();
    if (maybePromise instanceof Promise) {
        await maybePromise;
    }
} catch (e) {
    // ...
}
```